### PR TITLE
Validate pack2x16float

### PR DESCRIPTION
**Pull request by Jiawei-Shao**
Fixed: #3610 

<hr>

**Requirements for PR author:**

- [*] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [*] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [*] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [*] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const kGoodArgs = kArgCases['good'];
+const kReturnType = 'u32';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+g.test('args')
+  .desc(`Test compilation failure of ${kFn} with various numbers of and types of arguments`)
+  .params(u => u.combine('arg', keysOf(kArgCases)))
+  .fn(t => {
+    t.expectCompileResult(t.params.arg === 'good', `const c = ${kFn}${kArgCases[t.params.arg]};`);
```
Comment (ben-clayton) [inline src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts:38]: Would be good to test the pipeline-creation error for OOBs override expression case too.

Comment (ben-clayton) [reply src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts:38]: See: https://github.com/gpuweb/cts/blob/7e50fe3c028c3362af01380430472782e12a255f/src/webgpu/shader/validation/expression/call/builtin/const_override_validation.ts#L153

File: src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../../../common/util/data_tables.js';
+import { kValue } from '../../../../../../webgpu/util/constants.js';
+import { f32, vec2 } from '../../../../../util/conversion.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+import { validateConstOrOverrideBuiltinEval } from './const_override_validation.js';
+
+const kArgCases = {
+  good: '(vec2f())',
```
Comment (jzm-intel) [inline src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts:13]: Shall we add an abstract float case? Like
``` javascript
  good_vec2_af: '(vec2(0.0))',
```

**Review by jzm-intel — APPROVED** [review]
Looks good to me, thanks!

File: src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts (hunk 2 above)
Comment (Jiawei-Shao) [reply src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts:13]: Done

File: src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts (hunk 1 above)
Comment (Jiawei-Shao) [reply src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts]: Done

File: src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      value0 <= kValue.f16.positive.max &&
+      value1 >= kValue.f16.negative.min &&
+      value1 <= kValue.f16.positive.max;
+
+    validateConstOrOverrideBuiltinEval(
+      t,
+      kFn,
+      success,
+      [vec2(f32(value0), f32(value1))],
+      'override'
```
Comment (ben-clayton) [inline src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts:102]: Does this test need to only test `'override'` ? Could we not parameterize this to `['const', 'override']`, and then remove the OOBs cases from `kArgCases`?

Comment (Jiawei-Shao) [reply src/webgpu/shader/validation/expression/call/builtin/pack2x16float.spec.ts]: Done

**Review by dneto0 — APPROVED** [review]
Thanks!

**Pull request merged**